### ISSUE TITLE
DRTIO mastering broken on Sayma AMC

### ISSUE THREAD

**Issue by sbourdeauducq** [m-labs/artiq#1230]
* symptom: no link at transceiver level (no comma alignment)
* efd735a6ab3d2c4b1493fd2e032559d2c1bd525d does not have the bug, current master has it
* reverting high-resolution TTL clocking changes does not help
* both Master and MasterDAC variants are affected
* satellite operation is OK with and without high-resolution TTL clocking

**Comment by sbourdeauducq** [comment]
And trying to reduce Master to 1 channel results in this...
```
INFO: [Timing 38-35] Done setting XDC timing constraints.
INFO: [Timing 38-91] UpdateTimingParams: Speed grade: -1, Temperature grade: C, Delay Type: min_max.
INFO: [Timing 38-191] Multithreading enabled for timing update using a maximum of 8 CPUs
/opt/Xilinx/Vivado/2018.2/bin/loader: line 194: 31677 Segmentation fault      "$RDI_PROG" "$@"
```

**Comment by sbourdeauducq** [comment]
For some reason, it didn't crash the second time I ran it, and also the transceiver works correctly with only 1 channel.

**Comment by sbourdeauducq** [comment]
```
diff --git a/artiq/gateware/targets/sayma_amc.py b/artiq/gateware/targets/sayma_amc.py
index 6d680675..e6e1e27a 100755
--- a/artiq/gateware/targets/sayma_amc.py
+++ b/artiq/gateware/targets/sayma_amc.py
@@ -443,8 +443,7 @@ class Master(MiniSoC, AMPSoC):
         ]
         self.submodules.drtio_transceiver = gth_ultrascale.GTH(
             clock_pads=platform.request("si5324_clkout", 0),
-            data_pads=[platform.request("sfp", i) for i in range(2)] +
-                      [platform.request("rtm_gth", i) for i in range(8)],
+            data_pads=[platform.request("sfp", 0)],
             sys_clk_freq=self.clk_freq,
             rtio_clk_freq=rtio_clk_freq)
         self.csr_devices.append("drtio_transceiver")
@@ -455,7 +454,7 @@ class Master(MiniSoC, AMPSoC):
         drtioaux_csr_group = []
         drtioaux_memory_group = []
         drtio_cri = []
-        for i in range(10):
+        for i in range(1):
             core_name = "drtio" + str(i)
             coreaux_name = "drtioaux" + str(i)
             memory_name = "drtioaux" + str(i) + "_mem"
```

**Comment by sbourdeauducq** [comment]
no difference with and without high-resolution TTL clocking; single-channel master works in both cases.

**Comment by sbourdeauducq** [comment]
no difference with Vivado 2018.3

**Comment by sbourdeauducq** [comment]
Smells like a Xilinx bug. On a pristine master...
* setting the for loop range to 2 or 6 (from 10) results in a working DRTIO master
* setting the range to 8 crashes Vivado (deterministically) with 
``
ERROR: [Opt 31-67] Problem: A LUT3 cell in the design is missing a connection on input pin I2, which is used by the LUT equation. This pin has either been left unconnected in the design or the connection was removed due to the trimming of unused logic. The LUT cell name is: crimaster_state_reg[0]_i_3.
Resolution: Please review the preceding OPT INFO messages that detail what has been trimmed in the design to determine if the removal of unused logic is causing this error. If opt_design is being specified directly, it will need to be rerun with opt_design -verbose to generate detailed INFO messages about trimming.
``

**Comment by sbourdeauducq** [comment]
MasterDAC (with 2 or 3 links) is actually not affected, I had made a mistake when testing it the first time.

**Comment by sbourdeauducq** [comment]
GTH Ethernet seems to have made this situation worse - there are many other numbers of channels that now crash Vivado, and GTH Ethernet itself is often broken on the master variant (but works on masterdac).
Hopefully Xilinx will fix this in the next Vivado release, otherwise this will be another PITA for Metlino.

**Comment by sbourdeauducq** [comment]
Still a problem on Vivado 2019.1.

**Comment by sbourdeauducq** [comment]
Fixed in 2019.2.
